dashboard/db/migrate/20240122193916_create_ai_tutor_interactions.rb

```diff
@@ -0,0 +1,22 @@
+class CreateAiTutorInteractions < ActiveRecord::Migration[6.1]
+  def change
+    create_table :ai_tutor_interactions do |t|
+      t.integer :user_id, null: false, foreign_key: true
+      t.integer :level_id, foreign_key: true
+      t.integer :script_id, foreign_key: true
+      t.string :ai_model_version
+      t.string :type
+      t.string :project_id
+      t.string :project_version_id
+      t.text :prompt
+      t.string :status
+      t.text :ai_response
+
+      t.timestamps
+    end
+
+    add_index :ai_tutor_interactions, :user_id
```

Comment:
I've never used this syntax before, but I think for a composite key this should be...
```
add_index :ai_tutor_interactions, [:user_id, :level_id, :script_id], unique: false
```

Can someone with more Rails experience confirm?

Reply:
Right, these lines just add three single column indexes, but if you want a composite index, your syntax is correct (`column_name` can be an array)

Reply:
I like Emily's proposal for composite key. it looks like the index we use on the user_levels table, and I would naively expect access patterns for this table to be similar: https://github.com/code-dot-org/code-dot-org/blob/2866341a4f0e8b325d7ae972d42b67adc885ad52/dashboard/app/models/user_level.rb#L23

however it is important to keep in mind that this index only optimizes for the following types of queries:
- `where(user_id: ...)`
- `where(user_id: ..., level_id: ...)`
- `where(user_id: ..., level_id: ..., script_id: ...)`

if you find yourself querying by any of these things, you will want to add a new index:
- `where(level_id: ...)`
- `where(script_id: ...)`
- `where(level_id: ..., script_id: ...)`
- (any query at least including user_id will probably perform just fine for now)

ultimately, you should be (1) making a best-effort attempt now to anticipate what specific queries you will need to make today, and then (2) consider adding new indexes whenever you add new DB queries